### PR TITLE
style: Fix non-augmented-assignment (PLR6104)

### PR DESCRIPTION
**Pull request by echoix**
Ruff rule: https://docs.astral.sh/ruff/rules/non-augmented-assignment/

93 instances fixed, 3 more to discuss (they will get a ruff suggestion comment, as I left them out for now)

### CODE REVIEW

**Comment by ninsbl** [comment]
The changes look good. What is the reason you left the last three out?

File: scripts/i.spectral/i.spectral.py
Comment (echoix) [inline scripts/i.spectral/i.spectral.py:96]: Here it's not related to this issue specifically, but I don't undestand why enumerate is used here, but the count it provides isn't used here. (Enumerate is to go through the iterable, but also bundle the index in a tuple, by default the start argument is at 0). And the loop also keeps track of an `i` variable. Are we sure that the output that is written to the file is what is expected? It writes `i` as a `%d` (a number), but then `row` as a `%s` (a string). Without running it, it should then be the string representation of the `row`, that is in fact here a tuple of the count and the item of `what`. Is it expected? Where is it used?

File: python/grass/temporal/temporal_vector_algebra.py
Comment (echoix) [inline python/grass/temporal/temporal_vector_algebra.py:324]: These two changes here are the only ones that I'm a little less sure, as if you look at the doc page for this ruff rule, it shows an example of what can happen with mutable types, like lists, with an augmented assignment.
We aren't in the case that `resultlist` is assigned to a variable, but later augmented, but rather using `map_i.cmd_list` to augment `resultlist`, and then at the end `resultlist` is assigned to `map_i.cmd_list`.

I would just like a little second opinion on this.

File: scripts/i.spectral/i.spectral.py
Comment (ninsbl) [reply scripts/i.spectral/i.spectral.py:96]: To me it seems like a glitch, when the script was ported from shell to Python.

There are no examples or tests that can be used as a reference for how the textfile output is supposed to look like. With the current code it will look something like this:

```
1, (0, [1, 1, 255, 45, 65])
2, (1, [2, 2, 255, 233, 165])
3, (2, [3, 3, 125, 133, 155])
4, (3, [4, 4, 167, 34, 67])
```
That does not seem natural to me as I would expect rather CSV like text output... 

I would likely go for something like:

```
Path(output).write_text("\n".join([f"{i},{','.join(map(str, row))}"for i, row in enumerate(what, 1)]) + "\n")
```
Unless there is a reason for the current output format...

Comment (echoix) [reply scripts/i.spectral/i.spectral.py:96]: @neteler Do you have some more historical context, or an opinion here?

File: python/grass/temporal/temporal_vector_algebra.py
Comment (ninsbl) [reply python/grass/temporal/temporal_vector_algebra.py:324]: Seems fine to me. I ran some manual tests but could not find any issue with this augmentation...

Comment (echoix) [reply python/grass/temporal/temporal_vector_algebra.py:324]: I wasn't that scared too. Fine then!

File: scripts/i.spectral/i.spectral.py
Comment (echoix) [reply scripts/i.spectral/i.spectral.py:96]: The commit preceding the removal of the shell script version is https://github.com/OSGeo/grass/blob/3d692d1df5b57ddec09ca3ebd324b23e04baa03f/scripts/i.spectral/i.spectral

The author named in the docs is @neteler in 1998

The text file output was added by @YannChemin in 2015 in https://github.com/OSGeo/grass/commit/3107c3a8ccdf7ebf2745161594b233b222eea5ab

Comment (neteler) [reply scripts/i.spectral/i.spectral.py:96]: @ninsbl proposal sounds reasonable to me (and yeah, a output test would be very welcome).

Comment (wenzeslaus) [reply scripts/i.spectral/i.spectral.py:96]: I'm for using `i, row` with enumerate and outputting just that. That looks like the intention in the original code. I'm not sure how this compares to the one-liner.

Comment (echoix) [reply scripts/i.spectral/i.spectral.py:96]: Either way it will be in a separate PR, it is not something that blocks this PR

Comment (echoix) [reply scripts/i.spectral/i.spectral.py:96]: Created #4160 to track this

**Comment by echoix** [comment]
Waiting for an approval

**Review by ninsbl — APPROVED** [review]

**Pull request merged**
